src/Engine/Trigger/TriggerOnce.cpp

```diff
@@ -20,31 +20,31 @@ void TriggerOnce::OnRemain() {
 
 }
 
-std::string TriggerOnce::GetName() {
+const std::string& TriggerOnce::GetName() const {
     return name;
 }
 
-void TriggerOnce::SetName(std::string value) {
+void TriggerOnce::SetName(const std::string& value) {
     name = value;
 }
 
-std::string TriggerOnce::GetTargetFunction() {
+const std::string& TriggerOnce::GetTargetFunction() const {
     return targetFunction;
 }
 
-void TriggerOnce::SetTargetFunction(std::string value) {
+void TriggerOnce::SetTargetFunction(const std::string& value) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') The function 'SetTargetFunction' is never used. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)